File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        box = {}
        res = 0 
        l = 0 
        maxK = 0 
        for r in range(len(s)):
            maxK = max(maxK, box.get(s[r],0) + 1)
            while(r - l + 1 - maxK > k):
                box[s[l]] -= 1
                l += 1
            box[s[r]] = 1 + box.get(s[r], 0)
            res  = max(r - l + 1 , res)
            
        return res



        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> count (nums.begin(), nums.end());
        int res = 0 ;
        for(int num : count)
        {
            int length = 1; 
            while(count.find(num + length) != count.end())
            {
                length++;
            }
            res = max(length, res);
        }
        return res; 
        
    }
};
